Find a functionally equivalent code:com.android.systemui.statusbar.BaseStatusBar.visibilityChanged:COMMENT
Method Modifier: protected   
Comment:/**
 * The LEDs are turned o)ff when the notification panel is shown, even just a little bit.
 * This was added last-minute and is inconsistent with the way the rest of the notifications
 * are handled, because the notification isn't really cancelled.  The lights are just
 * turned off.  If any other notifications happen, the lights will turn back on.  Steve says
 * this is what he wants. (see bug 1131461)
 */

@@ -1,8 +1,15 @@
 {
     if (mPanelSlightlyVisible != visible) {
         mPanelSlightlyVisible = visible;
+        if (!visible) {
+            dismissPopups();
+        }
         try {
-            mBarService.onPanelRevealed();
+            if (visible) {
+                mBarService.onPanelRevealed();
+            } else {
+                mBarService.onPanelHidden();
+            }
         } catch (RemoteException ex) {
         // Won't fail unless the world has ended.
         }

